src/components/Proposal/ProposalSummary/ProposalSummaryOmenDisputeResolution.tsx

```diff
@@ -0,0 +1,168 @@
+import { IProposalState } from "@daostack/client";
+import classNames from "classnames";
+import { GenericSchemeInfo } from "genericSchemeRegistry";
+import { linkToEtherScan } from "lib/util";
+import * as React from "react";
+import * as css from "./ProposalSummary.scss";
+
+interface IProps {
+  genericSchemeInfo: GenericSchemeInfo;
+  detailView?: boolean;
+  proposal: IProposalState;
+  transactionModal?: boolean;
+}
+
+export default class ProposalSummaryOmenDisputeResolution extends React.Component<IProps, null> {
+
+  public render(): RenderOutput {
+    const { proposal, detailView, genericSchemeInfo, transactionModal } = this.props;
+    let decodedCallData: any;
+    try {
+      decodedCallData = genericSchemeInfo.decodeCallData(proposal.genericScheme.callData);
+    } catch(err) {
+      if (err.message.match(/no action matching/gi)) {
+        return <div>Error: {err.message} </div>;
+      } else {
+        throw err;
+      }
+    }
+    const action = decodedCallData.action;
+
+    const proposalSummaryClass = classNames({
+      [css.detailView]: detailView,
+      [css.transactionModal]: transactionModal,
+      [css.proposalSummary]: true,
+      [css.withDetails]: true,
+    });
+
+    switch (action.id) {
+      case "setMetaData":
```

Comment:
all of these look almost identical with only very minor changes in one part, maybe just use the case statement to change that one part so you dont have so much duplicate code?